### PR TITLE
feat: replacing afero.Fs with fs.FS

### PR DESCRIPTION
**Pull request by caarlos0**
BREAKING CHANGE: stop using `afero.Fs` and use the new `fs.FS` package instead.

Also improves windows support all over the place.

### CODE REVIEW

**Comment by caarlos0** [comment]
Not to be merged before go 1.16 stable is out...

**Comment by caarlos0** [comment]
setup-go action don't support go tip 😞

File: glob.go [hunk 1]
```diff
@@ -80,14 +83,21 @@ func toNixPath(path string) string {
 
 // Glob returns all files that match the given pattern in the current directory.
 func Glob(pattern string, opts ...OptFunc) ([]string, error) {
+	prefix := "./"
+	if strings.HasPrefix(pattern, stringSeparator) {
```
Comment (caarlos0) [inline glob.go:87]: I'm not entirely sure about this part yet... DirFS won't allow to start on `/`, so to keep the API I'm doing this little hack here... but maybe we can just declare it a breaking change?

**Comment by caarlos0** [comment]
cc/ @erikgeiser when you can, please take a look :)

File: glob.go [hunk 2]
```diff
@@ -74,20 +75,34 @@ func QuoteMeta(pattern string) string {
...
 
 // Glob returns all files that match the given pattern in the current directory.
+// If the given pattern indicates an absolute path, it will glob from `/`.
 func Glob(pattern string, opts ...OptFunc) ([]string, error) {
+	prefix := "./"
+	if strings.HasPrefix(pattern, stringSeparator) {
+		prefix = stringSeparator
+		opts = append(opts, func(opts *globOptions) {
+			opts.prefix = prefix
+		}, WithFs(os.DirFS(prefix)))
```
Comment (erikgeiser) [inline glob.go:90]: If I understand this correctly `globOptions.fs` will be a `os.DirFS` no matter what the user provides via `WithFS`, right?

File: glob.go (hunk 1 above)
Comment (erikgeiser) [reply glob.go:86]: What do you mean by that? `prefix = stringSeparator` and `WithFS(os.DirFS(prefix))` seems like you are doing exactly that.

As far as I understand the docs, `os.DirFS` has no concept of a current working directy. So normally I would expect the fs to be `os.DirFS("/")` so I can access everything. But if I open `file` I would expect it to open `/cwd/file` where `/cwd` is my current working directory. But instead, `os.DirFS("/") opens `/file`. So what we would need to keep track of is the current working directory right?

File: glob_test.go [hunk 3]
```diff
@@ -239,44 +258,44 @@ func TestGlob(t *testing.T) { // nolint:funlen
 
 	t.Run("match files in directories", func(t *testing.T) {
 		t.Parallel()
-		matches, err := Glob("/a/{b,c}", WithFs(testFs(t, []string{
-			"/a/b/d",
-			"/a/b/e/f",
-			"/a/c",
+		matches, err := Glob("a/{b,c}", WithFs(testFs(t, []string{
```
Comment (erikgeiser) [inline glob_test.go:261]: API-wise it should still be possible to do these tests with a leading `/`, right?

File: glob.go (hunk 1 above)
Comment (caarlos0) [reply glob.go:86]: fs.Open only accepts relative dirs

but I think this is indeed outside of the API. 

Before it was also only working on paths relative to current, and didn't accept absolute paths anyway (`/etc/foo/bar.conf` for example)

File: glob_test.go (hunk 3 above)
Comment (caarlos0) [reply glob_test.go:261]: never was actually, test was passing but shouldn't...

File: glob.go (hunk 2 above)
Comment (caarlos0) [reply glob.go:90]: yeah, I think this is wrong actually... 🤔

**Comment by caarlos0** [comment]
The problem with fs.FS is that it only accepts relative paths as arguments to Open, for example.

So, you'll do `os.DirFS("/")` and glob for `/foo/*.txt`, it wouldn't work, need to be `foo/*.txt` only, and if you do that, the result items will not have the `/` prefix, which looks weird.

Thinking about maybe creating an option `WithRootFS` maybe 🤔

File: glob_test.go (hunk 3 above)
Comment (erikgeiser) [reply glob_test.go:276]: Wow...
Well I would think that it should be possible to glob with absolute paths, though...

**Comment by erikgeiser** [comment]
Currently, I don't run Go 1.16 but I'll help investigate our options more actively as soon as Homebrew ships it.

**Comment by caarlos0** [comment]
Yeah there is no rush to fix this...

**Comment by caarlos0** [comment]
@erikgeiser I think its good to be reviewed again.

I make the Root path stuff an Option (and also added a "print options" option, for debugging et al).

Basically, by default, we start in current working dir (`.`) (as before). If the API user wants, they can use the `MaybeRootFS` option to detect that and work accordingly.

Prefix is needed in the case of "RootFS" because all results will be relative to the FS (in this case, `/`), so the results will be `etc/foo.conf` instead of `/etc/foo.conf`, for example, which may cause confusion.

**Comment by caarlos0** [comment]
It cost my sanity, but it is working on windows now.

**Pull request merged**